stackslib/src/chainstate/stacks/mod.rs

```diff
@@ -544,6 +551,35 @@ impl MultisigHashMode {
     }
 }
 
+impl OrderIndependentMultisigHashMode {
+    pub fn to_address_hash_mode(&self) -> AddressHashMode {
+        match *self {
+            OrderIndependentMultisigHashMode::P2SH => AddressHashMode::SerializeP2SH,
+            OrderIndependentMultisigHashMode::P2WSH => AddressHashMode::SerializeP2WSH,
+        }
+    }
+
+    pub fn from_address_hash_mode(hm: AddressHashMode) -> Option<OrderIndependentMultisigHashMode> {
+        match hm {
+            AddressHashMode::SerializeP2SH => Some(OrderIndependentMultisigHashMode::P2SH),
+            AddressHashMode::SerializeP2WSH => Some(OrderIndependentMultisigHashMode::P2WSH),
+            _ => None,
+        }
+    }
+
+    pub fn from_u8(n: u8) -> Option<OrderIndependentMultisigHashMode> {
```

Comment:
I know you're just following patterns here, but it's better to implement `TryFrom` for conversions that can fail, than make your own function and return an `Option`